File: backend/models/transaction.js
const mongoose = require('mongoose');

const TransactionSchema = new mongoose.Schema({
  bankName: {
    type: String,
    default: "ICICI Bank"
  },
  type: {
    type: String,
    default: "no type"
  },
  amount: {
    type: Number,
    default: 0.0,
    min: 1,
    max: 1000000
  },
  dateCreated: {
    type: Date,
    default: () => Date.now()
  },
  category: {
    type: String
  }
}, {
  collection: 'usersTransactions',
  timestamps: true
});

// **This is the key export**:
module.exports = mongoose.model('transaction', TransactionSchema);

File: spring-boot-jwtV1-master/src/main/java/com/Algonitefintech/fintech_security/model/Transaction.java
package com.Algonitefintech.fintech_security.model;

import lombok.*;
import java.util.Date;

import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;

import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "usersTransactions")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Transaction {
    @Id
    private ObjectId id;

    @Builder.Default
    private String bankName = "dummy bank";
    @Builder.Default
    private String type = "no type"; // e.g., "credited" or "debited"
    @Builder.Default
    private Double amount = 0.0;

    @Builder.Default
    private Date dateCreated = new Date();

    private String category;
}
